packages/web/app/src/lib/kit/helpers.ts

```diff
@@ -8,3 +8,20 @@ export const tryOr = <$PrimaryResult, $FallbackResult>(
     return fallback();
   }
 };
+
+export const oneOf = <type extends readonly unknown[]>(
+  ...guards: OneOfCheck<type>
+): ((value: unknown) => type[number]) => {
+  return (value: unknown) => {
+    for (const guard of guards) {
+      if (guard(value)) {
+        return value;
+      }
+    }
+    throw new Error(`Unexpected value received by oneOf: ${value}`);
```

Comment:
should this be a unique error class?

Reply:
Maybe! Its failure represents a defect not a normal error so we could encode that in a special class for sure.